### PR TITLE
Feature/wmt 1039/store logged in user in workload points

### PR DESCRIPTION
**Pull request by ddebella**

### CODE REVIEW

File: test/integration/services/data/test-get-workload-points.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ describe('services/data/get-workload-points', function () {
     return getExistingActiveWorkloadPoints().then(function (results) {
       expectedWorkloadPoints = results
       if (results === undefined) {
+        console.log('----------')
```
Comment (andrew-js-wright) [inline test/integration/services/data/test-get-workload-points.js]: Remove console.log

File: app/views/workload-points.html [hunk 2]
```diff
@@ -16,9 +16,7 @@
 
 <p>Workload points are set here and are used to calculate capacity points for all Offender Managers throughout the Workload Management Tool.</p>
 
-<!-- TODO: Include user. -->
-
-<p>Current Workload Points created by <span class="bold">test user</span> on <span class="bold">{{ wp.effectiveFrom }}</span></p> 
+<p>Current Workload Points created by <span class="bold">{{ wp.updatedByUserId }}</span> on <span class="bold">{{ wp.effectiveFrom }}</span></p>
```
Comment (andrew-js-wright) [inline app/views/workload-points.html]: As mentioned above, this should be a name by the time it's rendered so we could update the property name.

File: app/services/user-role-service.js [hunk 3]
```diff
@@ -56,9 +63,11 @@ module.exports.removeUserByUsername = function (username) {
 }
 
 var removeDomainFromUsername = function (username) {
```
Comment (andrew-js-wright) [inline app/services/user-role-service.js]: We could expose this as a public method, unit test it and use it in `app/authentication`.

File: app/authentication.js [hunk 4]
```diff
@@ -20,19 +21,23 @@ module.exports = function (app) {
...
     // Get the role for the user
-    getUserRoleByUsername(username).then(function (role) {
-      done(null, {
-        name: displayName,
-        username: username,
-        user_role: role.role,
-        nameID: nameID,
-        nameIDFormat: nameIDFormat
+    getUserByUsername(username).then(function (user) {
+      getUserRoleByUsername(username).then(function (role) {
```
Comment (andrew-js-wright) [inline app/authentication.js]: We could use the methods provided by the userRolesService for getting the user and role.

File: app/services/workload-points-service.js [hunk 5]
```diff
@@ -19,11 +20,17 @@ module.exports.getWorkloadPoints = function (id, organisationLevel) {
...
     }
-    result.title = breadcrumbs[0].title
-    result.subTitle = breadcrumbs[1].title
-    result.workloadPoints = results
-    result.breadcrumbs = breadcrumbs
-    return result
+    return userRoleService.getUserById(results.updatedByUserId)
+    .then(function (user) {
+      if (user !== undefined) {
+        results.updatedByUserId = user.name
```
[the remark this answers is not in the document]
Comment (ddebella) [reply app/services/workload-points-service.js]: Regarding the point about we are storing the name, we are storing the user_id into the workload points table. But when that is rendered to the view, we pass the actual name. I agree the `results.updatedByUserID = name` should be changed to something like `results.updatedBy = name`.

File: app/services/data/get-user-by-id.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+module.exports = function (id) {
+  if (id === undefined) {
+    throw new ReferenceError('User id is not defined')
+  }
+
+  return knex('users')
+    .where('id', id)
+    .select('id',
+            'username',
```
[the remark this answers is not in the document]
Comment (ddebella) [reply app/services/data/get-user-by-id.js]: Done.

File: app/views/workload-points.html (hunk 2 above)
Comment (ddebella) [reply app/views/workload-points.html]: Addressed above

File: app/services/user-role-service.js (hunk 3 above)
Comment (ddebella) [reply app/services/user-role-service.js]: done

File: app/authentication.js (hunk 4 above)
Comment (ddebella) [reply app/authentication.js]: Done

File: app/services/workload-points-service.js [hunk 7]
```diff
@@ -14,16 +15,27 @@ module.exports.getWorkloadPoints = function (id, organisationLevel) {
...
-    result.breadcrumbs = breadcrumbs
-    return result
+    return userRoleService.getUserById(workloadPoints.updatedByUserId)
+    .then(function (user) {
+      var updatedBy = workloadPoints.updatedByUserId // Default to the user id
+      if (user) {
+        if (user.name) { // If there is a valid use that
+          updatedBy = user.name
+        }
+        updatedBy = user.username
```
Comment (andrew-js-wright) [inline app/services/workload-points-service.js]: There needs to be an else here or the name will always get over written by the username.

**Pull request merged**